dice_ml/data_interfaces/public_data_interface.py

```diff
@@ -263,10 +266,15 @@ def get_indexes_of_features_to_vary(self, features_to_vary='all'):
     def from_dummies(self, data, prefix_sep='_'):
         """Gets the original data from dummy encoded data with k levels."""
         out = data.copy()
-        for l in self.categorical_feature_names:
+        for feat in self.categorical_feature_names:

```

Comment:
Can you write a comment in the code on the logic here? Cannot understand what the code is doing. Will be good to give an example, like educ_1, educ_2, and how it is processed.